### PR TITLE
Drop support for Node.js 6

### PR DESCRIPTION
**Pull request by b-admike**
Cherry pick some commits from #945 (see https://github.com/strongloop/loopback-next/pull/945#pullrequestreview-93198675). Related to #611.

## Checklist

- [x] `npm test` passes on your machine
- [x] New tests added or existing tests modified to cover all changes
- [x] Code conforms with the [style guide](http://loopback.io/doc/en/contrib/style-guide.html)
- [ ] Related API Documentation was updated
- [ ] Affected artifact templates in `packages/cli` were updated
- [ ] Affected example projects in `packages/example-*` were updated

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,9 +1,12 @@
 sudo: false
 language: node_js
 node_js:
-  - "6"
   - "8"
 
+os:
+  - linux
+  - osx
```
Comment (bajtos) [inline .travis.yml]: Can we leave `osx` support out of this patch please?

File: .travis.yml [hunk 2]
```diff
@@ -24,8 +27,8 @@ jobs:
       before_script: skip
       script: /bin/bash ./bin/lint-commits.sh
       after_success: skip
-      node_js:
-        - "8"
+      os:
+        - linux
```
Comment (bajtos) [inline .travis.yml]: Ditto.

File: packages/rest/src/parser.ts [hunk 3]
```diff
@@ -15,19 +15,10 @@ import {
...
-
-// tslint:disable:no-any
+// tslint:disable-next-line:no-any
 type MaybeBody = any | undefined;
-// tslint:enable:no-any
 
-const parseJsonBody: (req: ServerRequest) => Promise<MaybeBody> = promisify(
-  jsonBody,
-);
+const jsonBodyAsync = promisify(require('body/json'));
```
Comment (bajtos) [inline packages/rest/src/parser.ts]: Please preserve the name `parseJsonBody`, method/function names should start with a verb.

Also please verify that the compiler is correctly inferring the type of `promisify(require('body/json'))` and does not fall back to `any`. If it does not, then you need to preserve some of the type definitions above.

File: .travis.yml (hunk 1 above)
Comment (b-admike) [reply .travis.yml]: Yeah I was in the process of removing it :)

**Comment by bajtos** [comment]
> Oh, hang on. I just realized that we need to remove build:dist6 scripts from our package.json files. There is no point in transpiling for ES2015 (Node.js 6.x) when we no longer support it.

On the second thought - when we remove `build:dist6`, then we must change package-level `index.js` files too, because we can no longer forward to `dist6`, at which point we may be too close to #945? 

I am no longer sure what's the right approach here. Does anybody else have any opinion?

The only important thing that remains is to ensure the commit message landing these changes contains `BREAKING CHANGE` footer, so that our release notes explicitly mention this change. See https://conventionalcommits.org/

For example:

```
BREAKING CHANGE: Support for Node.js version lower than 8.0 has been dropped.
Please upgrade to the latest Node.js 8.x LTS version.
```

**Comment by virkt25** [comment]
Need to add a `Test` stage for Travis because right now only thing running is Commit Lint because not having a `os` field or multiple versions of `node` means there is no longer a default stage Travis generates. Right now the only thing Travis is doing is commit message linting.

**Comment by b-admike** [comment]
@bajtos re: https://github.com/strongloop/loopback-next/pull/948#issuecomment-362293637, perhaps we can add https://github.com/strongloop/loopback-next/pull/945/commits/4a62b0eeb999bf8e0b2c929c20291fbea90c134e to this PR instead of the other?

**Comment by kjdelisle** [comment]
It does suck that we're losing almost 1% of code coverage because of this change, but that's more indicative of a problem we already had than one we're creating.

**Comment by b-admike** [comment]
Yeah I'm trying to figure out why the sudden drop of coverage :(.

**Comment by b-admike** [comment]
@virkt25 @kjdelisle Travis should be good now that we landed #949.

**Comment by bajtos** [comment]
> t does suck that we're losing almost 1% of code coverage because of this change, but that's more indicative of a problem we already had than one we're creating.

I checked Coveralls report. The only relevant change in coverage numbers I can see are for [`rest/src/parser.ts`](https://coveralls.io/builds/15334119/source?filename=packages%2Frest%2Fsrc%2Fparser.ts).

I suspect our coverage numbers by me skewed by the fact that we are reporting coverage for each Node.js platform we are building. Because the code emitted by `tsc` is different, especially for Node.js 6.x with no support for `async`/`await`, I suspect the coverage transformed via sourcemaps could provide slightly different data on each Node.js version.

I personally don't worry about the exact coverage number, especially when it's well over 90%. For me, coverage data is just a tool helping us to spot parts of code that are not sufficiently covered by tests.

The Travis setup seems to work wonderfully 👍  Let's get this landed! 🎉 

screen shot 2018-02-02 at 10 26 29

File: package.json [hunk 4]
```diff
@@ -6,7 +6,7 @@
   },
   "version": "4.0.0-alpha.1",
   "engines": {
-    "node": ">=6"
+    "node": ">=8"
```
Comment (richardpringle) [inline package.json]: 🎉

**Pull request merged**